### ISSUE TITLE
Crash since 0.9.0 (bs-platform 7.3.x issue)

### ISSUE THREAD

**Issue by sync** [zth/reason-relay#59]
Still looking at what's happening but here it's what i have so far

```
SSR fetch error TypeError: Cannot read property 'undefined' of undefined
    at Object.traverser (/Users/anthonymittaz/Projects/Homepass/homepass-meet/meet-app/.next/server/static/development/pages/room.js:14912:38)
    at Module._convertObj (/Users/anthonymittaz/Projects/Homepass/homepass-meet/meet-app/.next/server/static/development/pages/room.js:13901:58)
    at convertResponse (/Users/anthonymittaz/Projects/Homepass/homepass-meet/meet-app/.next/server/static/development/pages/room.js:15720:88)
    at Module._1 (/Users/anthonymittaz/Projects/Homepass/homepass-meet/meet-app/.next/server/static/development/pages/room.js:8222:12)
    at /Users/anthonymittaz/Projects/Homepass/homepass-meet/meet-app/.next/server/static/development/pages/room.js:14657:138
    at runMicrotasks (<anonymous>)
```

<img width="941" alt="Screen Shot 2020-05-18 at 8 38 48 am" src="https://user-images.githubusercontent.com/21725/82161872-a7e4a880-98e3-11ea-869a-25be725bf9e4.png">

**Comment by sync** [comment]
relay and react version
```
    "react": "0.0.0-experimental-33c3af284",
    "react-dom": "0.0.0-experimental-33c3af284",
    "react-relay": "0.0.0-experimental-895a6fe0",
```
relay compiler and friends are at `9.1.0`

**Comment by zth** [comment]
Hi! Could you paste the generated file for the operation it's trying to use? And did you re-run the compiler?

**Comment by zth** [comment]
Also I'd downgrade to Relay 9 from 9.1 since that's what I'm testing ReasonRelay with right now. Just to be sure it's nothing like that.

**Comment by sync** [comment]
yes reran the relay compiler after the update

**Comment by sync** [comment]
```


import * as ReasonRelay from "reason-relay/src/ReasonRelay.bs.js";

function makeRefetchVariables(spaceId, otherParticipantId, skipOtherParticipant, currentParticipantId, skipCurrentParticipant, param) {
  return {
          spaceId: spaceId,
          otherParticipantId: otherParticipantId,
          skipOtherParticipant: skipOtherParticipant,
          currentParticipantId: currentParticipantId,
          skipCurrentParticipant: skipCurrentParticipant
        };
}

var Types = {
  makeRefetchVariables: makeRefetchVariables
};

var responseConverter = ({"__root":{"space":{"n":"","f":""},"space_account":{"n":""},"space_account_name":{"n":""},"space_branding":{"n":""},"space_branding_kiosk":{"n":""},"space_branding_kiosk_logoUrl":{"n":""},"space_branding_kiosk_listingAgents":{"n":"","na":""},"space_branding_kiosk_listingAgents_name":{"n":""},"space_branding_kiosk_listingAgents_imageUrl":{"n":""},"":{"f":""}}});

function convertResponse(v) {
  return ReasonRelay._convertObj(v, responseConverter, undefined, undefined);
}

var variablesConverter = ({});

function convertVariables(v) {
  return ReasonRelay._convertObj(v, variablesConverter, undefined, undefined);
}

var Internal = {
  responseConverter: responseConverter,
  responseConverterMap: undefined,
  convertResponse: convertResponse,
  variablesConverter: variablesConverter,
  variablesConverterMap: undefined,
  convertVariables: convertVariables
};

function makeVariables(spaceId, otherParticipantId, skipOtherParticipant, currentParticipantId, skipCurrentParticipant) {
  return {
          spaceId: spaceId,
          otherParticipantId: otherParticipantId,
          skipOtherParticipant: skipOtherParticipant,
          currentParticipantId: currentParticipantId,
          skipCurrentParticipant: skipCurrentParticipant
        };
}

var Utils = {
  makeVariables: makeVariables
};

var node = ((function(){
var v0 = [
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "spaceId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "otherParticipantId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "skipOtherParticipant",
    "type": "Boolean!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "currentParticipantId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "skipCurrentParticipant",
    "type": "Boolean!"
  }
],
v1 = [
  {
    "kind": "Variable",
    "name": "id",
    "variableName": "spaceId"
  }
],
v2 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "__typename",
  "storageKey": null
},
v3 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "name",
  "storageKey": null
},
v4 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "testAccount",
  "storageKey": null
},
v5 = {
  "alias": null,
  "args": null,
  "concreteType": "SpaceBrandType",
  "kind": "LinkedField",
  "name": "branding",
  "plural": false,
  "selections": [
    {
      "alias": null,
      "args": null,
      "concreteType": "KioskBrandType",
      "kind": "LinkedField",
      "name": "kiosk",
      "plural": false,
      "selections": [
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "primaryColor",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "accentColor",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "logoUrl",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "concreteType": "listingAgent",
          "kind": "LinkedField",
          "name": "listingAgents",
          "plural": true,
          "selections": [
            (v3/*: any*/),
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "imageUrl",
              "storageKey": null
            }
          ],
          "storageKey": null
        }
      ],
      "storageKey": null
    }
  ],
  "storageKey": null
},
v6 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "id",
  "storageKey": null
},
v7 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "fullName",
  "storageKey": null
},
v8 = [
  (v3/*: any*/)
],
v9 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "mobile",
  "storageKey": null
},
v10 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "email",
  "storageKey": null
},
v11 = [
  (v3/*: any*/),
  (v9/*: any*/),
  (v10/*: any*/)
];
return {
  "fragment": {
    "argumentDefinitions": (v0/*: any*/),
    "kind": "Fragment",
    "metadata": null,
    "name": "roomQuery",
    "selections": [
      {
        "alias": "space",
        "args": (v1/*: any*/),
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          (v2/*: any*/),
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "concreteType": "Account",
                "kind": "LinkedField",
                "name": "account",
                "plural": false,
                "selections": [
                  (v3/*: any*/),
                  (v4/*: any*/)
                ],
                "storageKey": null
              },
              (v5/*: any*/)
            ],
            "type": "Space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomSpaceInfo_space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomJoinAttendee_space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomJoinDevice_space"
          }
        ],
        "storageKey": null
      },
      {
        "args": null,
        "kind": "FragmentSpread",
        "name": "RoomWaitingForOther_query"
      },
      {
        "args": null,
        "kind": "FragmentSpread",
        "name": "RoomJoinAttendeeCheckin_query"
      }
    ],
    "type": "Query"
  },
  "kind": "Request",
  "operation": {
    "argumentDefinitions": (v0/*: any*/),
    "kind": "Operation",
    "name": "roomQuery",
    "selections": [
      {
        "alias": "space",
        "args": (v1/*: any*/),
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          (v2/*: any*/),
          (v6/*: any*/),
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "concreteType": "Account",
                "kind": "LinkedField",
                "name": "account",
                "plural": false,
                "selections": [
                  (v3/*: any*/),
                  (v4/*: any*/),
                  (v6/*: any*/),
                  {
                    "alias": null,
                    "args": null,
                    "kind": "ScalarField",
                    "name": "accountId",
                    "storageKey": null
                  }
                ],
                "storageKey": null
              },
              (v5/*: any*/),
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "title",
                "storageKey": null
              },
              {
                "alias": "metaImageUrl",
                "args": null,
                "kind": "ScalarField",
                "name": "imageUrl",
                "storageKey": null
              },
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "url",
                "storageKey": null
              },
              {
                "alias": "welcomeImageUrl",
                "args": [
                  {
                    "kind": "Literal",
                    "name": "dpr",
                    "value": 2
                  },
                  {
                    "kind": "Literal",
                    "name": "width",
                    "value": 400
                  }
                ],
                "kind": "ScalarField",
                "name": "imageUrl",
                "storageKey": "imageUrl(dpr:2,width:400)"
              }
            ],
            "type": "Space"
          }
        ],
        "storageKey": null
      },
      {
        "condition": "skipOtherParticipant",
        "kind": "Condition",
        "passingValue": false,
        "selections": [
          {
            "alias": "otherParticipant",
            "args": [
              {
                "kind": "Variable",
                "name": "id",
                "variableName": "otherParticipantId"
              }
            ],
            "concreteType": null,
            "kind": "LinkedField",
            "name": "node",
            "plural": false,
            "selections": [
              (v2/*: any*/),
              (v6/*: any*/),
              {
                "kind": "InlineFragment",
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "Contact",
                    "kind": "LinkedField",
                    "name": "contact",
                    "plural": false,
                    "selections": [
                      (v7/*: any*/),
                      (v6/*: any*/)
                    ],
                    "storageKey": null
                  }
                ],
                "type": "SpaceContact"
              },
              {
                "kind": "InlineFragment",
                "selections": (v8/*: any*/),
                "type": "Attendee"
              },
              {
                "kind": "InlineFragment",
                "selections": (v8/*: any*/),
                "type": "User"
              }
            ],
            "storageKey": null
          }
        ]
      },
      {
        "condition": "skipCurrentParticipant",
        "kind": "Condition",
        "passingValue": false,
        "selections": [
          {
            "alias": "currentParticipant",
            "args": [
              {
                "kind": "Variable",
                "name": "id",
                "variableName": "currentParticipantId"
              }
            ],
            "concreteType": null,
            "kind": "LinkedField",
            "name": "node",
            "plural": false,
            "selections": [
              (v2/*: any*/),
              (v6/*: any*/),
              {
                "kind": "InlineFragment",
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "Contact",
                    "kind": "LinkedField",
                    "name": "contact",
                    "plural": false,
                    "selections": [
                      (v7/*: any*/),
                      (v9/*: any*/),
                      (v10/*: any*/),
                      (v6/*: any*/)
                    ],
                    "storageKey": null
                  }
                ],
                "type": "SpaceContact"
              },
              {
                "kind": "InlineFragment",
                "selections": (v11/*: any*/),
                "type": "Attendee"
              },
              {
                "kind": "InlineFragment",
                "selections": (v11/*: any*/),
                "type": "User"
              }
            ],
            "storageKey": null
          }
        ]
      }
    ]
  },
  "params": {
    "id": null,
    "metadata": {},
    "name": "roomQuery",
    "operationKind": "query",
    "text": "query roomQuery(\n  $spaceId: ID!\n  $otherParticipantId: ID!\n  $skipOtherParticipant: Boolean!\n  $currentParticipantId: ID!\n  $skipCurrentParticipant: Boolean!\n) {\n  space: node(id: $spaceId) {\n    __typename\n    ... on Space {\n      account {\n        name\n        testAccount\n        id\n      }\n      branding {\n        kiosk {\n          primaryColor\n          accentColor\n          logoUrl\n          listingAgents {\n            name\n            imageUrl\n          }\n        }\n      }\n    }\n    ...RoomSpaceInfo_space\n    ...RoomJoinAttendee_space\n    ...RoomJoinDevice_space\n    id\n  }\n  ...RoomWaitingForOther_query\n  ...RoomJoinAttendeeCheckin_query\n}\n\nfragment RoomJoinAttendeeCheckin_query on Query {\n  currentParticipant: node(id: $currentParticipantId) @skip(if: $skipCurrentParticipant) {\n    __typename\n    ... on SpaceContact {\n      contact {\n        fullName\n        mobile\n        email\n        id\n      }\n    }\n    ... on Attendee {\n      name\n      mobile\n      email\n    }\n    ... on User {\n      name\n      mobile\n      email\n    }\n    id\n  }\n}\n\nfragment RoomJoinAttendeeCheckin_space on Space {\n  account {\n    accountId\n    id\n  }\n}\n\nfragment RoomJoinAttendeeWelcome_space on Space {\n  title\n  welcomeImageUrl: imageUrl(width: 400, dpr: 2)\n}\n\nfragment RoomJoinAttendee_space on Space {\n  ...RoomJoinAttendeeWelcome_space\n  ...RoomJoinAttendeeCheckin_space\n}\n\nfragment RoomJoinDevice_space on Space {\n  title\n}\n\nfragment RoomSpaceInfo_space on Space {\n  title\n  metaImageUrl: imageUrl\n  url\n}\n\nfragment RoomWaitingForOther_query on Query {\n  otherParticipant: node(id: $otherParticipantId) @skip(if: $skipOtherParticipant) {\n    __typename\n    ... on SpaceContact {\n      contact {\n        fullName\n        id\n      }\n    }\n    ... on Attendee {\n      name\n    }\n    ... on User {\n      name\n    }\n    id\n  }\n}\n"
  }
};
})());

var include = ReasonRelay.MakePreloadQuery({
      query: node,
      convertVariables: convertVariables
    });

var preload = include.preload;

var preloadTokenToObservable = include.preloadTokenToObservable;

var preloadTokenToPromise = include.preloadTokenToPromise;

export {
  Types ,
  Internal ,
  Utils ,
  node ,
  preload ,
  preloadTokenToObservable ,
  preloadTokenToPromise ,
  
}
/* responseConverter Not a pure module */
```

**Comment by sync** [comment]
i'll try to downgrade to 9.0 to see if it makes any difference, but it worked before on 0.8.3 with the same relay versions

**Comment by zth** [comment]
Yeah sure, it's mostly to rule it out completely.

Could you also paste the Reason of that generated module? And your bs-platform version.

**Comment by sync** [comment]
```
/* @generated */

module Types = {
  type response_space_branding_kiosk_listingAgents = {
    name: option(string),
    imageUrl: option(string),
  };
  type response_space_branding_kiosk = {
    primaryColor: string,
    accentColor: string,
    logoUrl: option(string),
    listingAgents:
      option(array(option(response_space_branding_kiosk_listingAgents))),
  };
  type response_space_branding = {
    kiosk: option(response_space_branding_kiosk),
  };
  type response_space_account = {
    name: option(string),
    testAccount: bool,
  };
  type response_space = {
    __typename: string,
    account: option(response_space_account),
    branding: option(response_space_branding),
    getFragmentRefs:
      unit =>
      {
        .
        "__$fragment_ref__RoomSpaceInfo_space": RoomSpaceInfo_space_graphql.t,
        "__$fragment_ref__RoomJoinAttendee_space": RoomJoinAttendee_space_graphql.t,
        "__$fragment_ref__RoomJoinDevice_space": RoomJoinDevice_space_graphql.t,
      },
  };

  type response = {
    space: option(response_space),
    getFragmentRefs:
      unit =>
      {
        .
        "__$fragment_ref__RoomWaitingForOther_query": RoomWaitingForOther_query_graphql.t,
        "__$fragment_ref__RoomJoinAttendeeCheckin_query": RoomJoinAttendeeCheckin_query_graphql.t,
      },
  };
  type refetchVariables = {
    spaceId: option(string),
    otherParticipantId: option(string),
    skipOtherParticipant: option(bool),
    currentParticipantId: option(string),
    skipCurrentParticipant: option(bool),
  };
  let makeRefetchVariables =
      (
        ~spaceId=?,
        ~otherParticipantId=?,
        ~skipOtherParticipant=?,
        ~currentParticipantId=?,
        ~skipCurrentParticipant=?,
        (),
      )
      : refetchVariables => {
    spaceId,
    otherParticipantId,
    skipOtherParticipant,
    currentParticipantId,
    skipCurrentParticipant,
  };
  type variables = {
    spaceId: string,
    otherParticipantId: string,
    skipOtherParticipant: bool,
    currentParticipantId: string,
    skipCurrentParticipant: bool,
  };
};

module Internal = {
  type responseRaw;
  let responseConverter: Js.Dict.t(Js.Dict.t(Js.Dict.t(string))) = [%raw
    {json| {"__root":{"space":{"n":"","f":""},"space_account":{"n":""},"space_account_name":{"n":""},"space_branding":{"n":""},"space_branding_kiosk":{"n":""},"space_branding_kiosk_logoUrl":{"n":""},"space_branding_kiosk_listingAgents":{"n":"","na":""},"space_branding_kiosk_listingAgents_name":{"n":""},"space_branding_kiosk_listingAgents_imageUrl":{"n":""},"":{"f":""}}} |json}
  ];
  let responseConverterMap = ();
  let convertResponse = v =>
    v
    ->ReasonRelay._convertObj(
        responseConverter,
        responseConverterMap,
        Js.undefined,
      );

  let variablesConverter: Js.Dict.t(Js.Dict.t(Js.Dict.t(string))) = [%raw
    {json| {} |json}
  ];
  let variablesConverterMap = ();
  let convertVariables = v =>
    v
    ->ReasonRelay._convertObj(
        variablesConverter,
        variablesConverterMap,
        Js.undefined,
      );
};

type preloadToken;

module Utils = {
  open Types;
  let makeVariables =
      (
        ~spaceId,
        ~otherParticipantId,
        ~skipOtherParticipant,
        ~currentParticipantId,
        ~skipCurrentParticipant,
      )
      : variables => {
    spaceId,
    otherParticipantId,
    skipOtherParticipant,
    currentParticipantId,
    skipCurrentParticipant,
  };
};

type operationType = ReasonRelay.queryNode;

let node: operationType = [%raw
  {json| (function(){
var v0 = [
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "spaceId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "otherParticipantId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "skipOtherParticipant",
    "type": "Boolean!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "currentParticipantId",
    "type": "ID!"
  },
  {
    "defaultValue": null,
    "kind": "LocalArgument",
    "name": "skipCurrentParticipant",
    "type": "Boolean!"
  }
],
v1 = [
  {
    "kind": "Variable",
    "name": "id",
    "variableName": "spaceId"
  }
],
v2 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "__typename",
  "storageKey": null
},
v3 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "name",
  "storageKey": null
},
v4 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "testAccount",
  "storageKey": null
},
v5 = {
  "alias": null,
  "args": null,
  "concreteType": "SpaceBrandType",
  "kind": "LinkedField",
  "name": "branding",
  "plural": false,
  "selections": [
    {
      "alias": null,
      "args": null,
      "concreteType": "KioskBrandType",
      "kind": "LinkedField",
      "name": "kiosk",
      "plural": false,
      "selections": [
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "primaryColor",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "accentColor",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "logoUrl",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "concreteType": "listingAgent",
          "kind": "LinkedField",
          "name": "listingAgents",
          "plural": true,
          "selections": [
            (v3/*: any*/),
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "imageUrl",
              "storageKey": null
            }
          ],
          "storageKey": null
        }
      ],
      "storageKey": null
    }
  ],
  "storageKey": null
},
v6 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "id",
  "storageKey": null
},
v7 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "fullName",
  "storageKey": null
},
v8 = [
  (v3/*: any*/)
],
v9 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "mobile",
  "storageKey": null
},
v10 = {
  "alias": null,
  "args": null,
  "kind": "ScalarField",
  "name": "email",
  "storageKey": null
},
v11 = [
  (v3/*: any*/),
  (v9/*: any*/),
  (v10/*: any*/)
];
return {
  "fragment": {
    "argumentDefinitions": (v0/*: any*/),
    "kind": "Fragment",
    "metadata": null,
    "name": "roomQuery",
    "selections": [
      {
        "alias": "space",
        "args": (v1/*: any*/),
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          (v2/*: any*/),
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "concreteType": "Account",
                "kind": "LinkedField",
                "name": "account",
                "plural": false,
                "selections": [
                  (v3/*: any*/),
                  (v4/*: any*/)
                ],
                "storageKey": null
              },
              (v5/*: any*/)
            ],
            "type": "Space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomSpaceInfo_space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomJoinAttendee_space"
          },
          {
            "args": null,
            "kind": "FragmentSpread",
            "name": "RoomJoinDevice_space"
          }
        ],
        "storageKey": null
      },
      {
        "args": null,
        "kind": "FragmentSpread",
        "name": "RoomWaitingForOther_query"
      },
      {
        "args": null,
        "kind": "FragmentSpread",
        "name": "RoomJoinAttendeeCheckin_query"
      }
    ],
    "type": "Query"
  },
  "kind": "Request",
  "operation": {
    "argumentDefinitions": (v0/*: any*/),
    "kind": "Operation",
    "name": "roomQuery",
    "selections": [
      {
        "alias": "space",
        "args": (v1/*: any*/),
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          (v2/*: any*/),
          (v6/*: any*/),
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "concreteType": "Account",
                "kind": "LinkedField",
                "name": "account",
                "plural": false,
                "selections": [
                  (v3/*: any*/),
                  (v4/*: any*/),
                  (v6/*: any*/),
                  {
                    "alias": null,
                    "args": null,
                    "kind": "ScalarField",
                    "name": "accountId",
                    "storageKey": null
                  }
                ],
                "storageKey": null
              },
              (v5/*: any*/),
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "title",
                "storageKey": null
              },
              {
                "alias": "metaImageUrl",
                "args": null,
                "kind": "ScalarField",
                "name": "imageUrl",
                "storageKey": null
              },
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "url",
                "storageKey": null
              },
              {
                "alias": "welcomeImageUrl",
                "args": [
                  {
                    "kind": "Literal",
                    "name": "dpr",
                    "value": 2
                  },
                  {
                    "kind": "Literal",
                    "name": "width",
                    "value": 400
                  }
                ],
                "kind": "ScalarField",
                "name": "imageUrl",
                "storageKey": "imageUrl(dpr:2,width:400)"
              }
            ],
            "type": "Space"
          }
        ],
        "storageKey": null
      },
      {
        "condition": "skipOtherParticipant",
        "kind": "Condition",
        "passingValue": false,
        "selections": [
          {
            "alias": "otherParticipant",
            "args": [
              {
                "kind": "Variable",
                "name": "id",
                "variableName": "otherParticipantId"
              }
            ],
            "concreteType": null,
            "kind": "LinkedField",
            "name": "node",
            "plural": false,
            "selections": [
              (v2/*: any*/),
              (v6/*: any*/),
              {
                "kind": "InlineFragment",
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "Contact",
                    "kind": "LinkedField",
                    "name": "contact",
                    "plural": false,
                    "selections": [
                      (v7/*: any*/),
                      (v6/*: any*/)
                    ],
                    "storageKey": null
                  }
                ],
                "type": "SpaceContact"
              },
              {
                "kind": "InlineFragment",
                "selections": (v8/*: any*/),
                "type": "Attendee"
              },
              {
                "kind": "InlineFragment",
                "selections": (v8/*: any*/),
                "type": "User"
              }
            ],
            "storageKey": null
          }
        ]
      },
      {
        "condition": "skipCurrentParticipant",
        "kind": "Condition",
        "passingValue": false,
        "selections": [
          {
            "alias": "currentParticipant",
            "args": [
              {
                "kind": "Variable",
                "name": "id",
                "variableName": "currentParticipantId"
              }
            ],
            "concreteType": null,
            "kind": "LinkedField",
            "name": "node",
            "plural": false,
            "selections": [
              (v2/*: any*/),
              (v6/*: any*/),
              {
                "kind": "InlineFragment",
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "Contact",
                    "kind": "LinkedField",
                    "name": "contact",
                    "plural": false,
                    "selections": [
                      (v7/*: any*/),
                      (v9/*: any*/),
                      (v10/*: any*/),
                      (v6/*: any*/)
                    ],
                    "storageKey": null
                  }
                ],
                "type": "SpaceContact"
              },
              {
                "kind": "InlineFragment",
                "selections": (v11/*: any*/),
                "type": "Attendee"
              },
              {
                "kind": "InlineFragment",
                "selections": (v11/*: any*/),
                "type": "User"
              }
            ],
            "storageKey": null
          }
        ]
      }
    ]
  },
  "params": {
    "id": null,
    "metadata": {},
    "name": "roomQuery",
    "operationKind": "query",
    "text": "query roomQuery(\n  $spaceId: ID!\n  $otherParticipantId: ID!\n  $skipOtherParticipant: Boolean!\n  $currentParticipantId: ID!\n  $skipCurrentParticipant: Boolean!\n) {\n  space: node(id: $spaceId) {\n    __typename\n    ... on Space {\n      account {\n        name\n        testAccount\n        id\n      }\n      branding {\n        kiosk {\n          primaryColor\n          accentColor\n          logoUrl\n          listingAgents {\n            name\n            imageUrl\n          }\n        }\n      }\n    }\n    ...RoomSpaceInfo_space\n    ...RoomJoinAttendee_space\n    ...RoomJoinDevice_space\n    id\n  }\n  ...RoomWaitingForOther_query\n  ...RoomJoinAttendeeCheckin_query\n}\n\nfragment RoomJoinAttendeeCheckin_query on Query {\n  currentParticipant: node(id: $currentParticipantId) @skip(if: $skipCurrentParticipant) {\n    __typename\n    ... on SpaceContact {\n      contact {\n        fullName\n        mobile\n        email\n        id\n      }\n    }\n    ... on Attendee {\n      name\n      mobile\n      email\n    }\n    ... on User {\n      name\n      mobile\n      email\n    }\n    id\n  }\n}\n\nfragment RoomJoinAttendeeCheckin_space on Space {\n  account {\n    accountId\n    id\n  }\n}\n\nfragment RoomJoinAttendeeWelcome_space on Space {\n  title\n  welcomeImageUrl: imageUrl(width: 400, dpr: 2)\n}\n\nfragment RoomJoinAttendee_space on Space {\n  ...RoomJoinAttendeeWelcome_space\n  ...RoomJoinAttendeeCheckin_space\n}\n\nfragment RoomJoinDevice_space on Space {\n  title\n}\n\nfragment RoomSpaceInfo_space on Space {\n  title\n  metaImageUrl: imageUrl\n  url\n}\n\nfragment RoomWaitingForOther_query on Query {\n  otherParticipant: node(id: $otherParticipantId) @skip(if: $skipOtherParticipant) {\n    __typename\n    ... on SpaceContact {\n      contact {\n        fullName\n        id\n      }\n    }\n    ... on Attendee {\n      name\n    }\n    ... on User {\n      name\n    }\n    id\n  }\n}\n"
  }
};
})() |json}
];

include ReasonRelay.MakePreloadQuery({
  type variables = Types.variables;
  type queryPreloadToken = preloadToken;
  let query = node;
  let convertVariables = Internal.convertVariables;
});
```

**Comment by sync** [comment]
```
    "bs-platform": "7.3.2",
```

**Comment by sync** [comment]
will run it on relay in just a bit

**Comment by sync** [comment]
same error with 9.0.0

**Comment by sync** [comment]
also using latest react, react, react dom  and relay experimental versions, will revert to previous version

**Comment by zth** [comment]
I might have an idea of what's causing this. Could you try downgrading `bs-platform` to `7.2.x` just to try if that works?

**Comment by sync** [comment]
yeah every time i see issues with undefined i am thinking it's 7.3

**Comment by sync** [comment]
yes it's 7.3, with bsb 7.2.2 i don't have that issue

**Comment by zth** [comment]
Yeah, it's 7.3 unfortunately. I have a good idea of what it is and how to fix this, but I won't have time until tonight to fix + cut new release. I'll get back to you as soon as I've had a chance to look at it properly 👍

**Comment by sync** [comment]
no rush, i have a work around

**Comment by sync** [comment]
that's the diff 
```
function convertVariables(v) {
  return ReasonRelay._convertObj(v, variablesConverter, undefined, undefined);
  return ReasonRelay._convertObj(v, variablesConverter, /* () */0, undefined);
}
```

**Comment by zth** [comment]
Yup! This is a very very weird JS-related issue 😄 

Unfortunately coming from a part of the code base that's not properly typed with Reason or TS. I should probably try and re-rewrite that at some point, but starting out I'll just fix it to handle both representations.

**Comment by zth** [comment]
I suspect this type of thing is the root of this bug:

```
0["test"]
> undefined
undefined["test]
> VM162:1 Uncaught SyntaxError: Invalid or unexpected token
```

**Comment by sync** [comment]
oh boy

**Comment by zth** [comment]
Fixed in https://github.com/zth/reason-relay/commit/a28043b438ba5b76336472144e86028eaa455453 and released in `0.9.1`. Please give it a spin and re-open if it doesn't work out for you.

**Comment by sync** [comment]
worked thanks
